lib/src/baseFormValidation.ts

```diff
@@ -13,25 +16,42 @@ interface FormValidation {
   isValidationInProgress(): boolean;
   isFormDirty(): boolean;
   isFormPristine(): boolean;
-  addFieldValidation(key: string, validation: (value: string, vm: any) => FieldValidationResult, filter?: any): FormValidation;
-  addFieldValidationAsync(key: string, validation: (value: string, vm: any) => Promise<FieldValidationResult>, filter?: any): FormValidation;
 }
 
 export class BaseFormValidation implements FormValidation {
   private validationEngine: ValidationEngine;
+
   constructor(validationConstraints: ValidationConstraints) {
     this.validationEngine = new ValidationEngine();
     this.parseValidationConstraints(validationConstraints);
   }
 
   private parseValidationConstraints(validationConstraints: ValidationConstraints) {
     if (validationConstraints && typeof validationConstraints === 'object') {
-      if (validationConstraints.global && validationConstraints.global instanceof Array) {
-        this.addFormValidationFunctions(validationConstraints.global);
+      const { global, fields } = validationConstraints;
+      if (global && global instanceof Array) {
+        this.addFormValidationFunctions(global);
+      }
+      if (fields && typeof fields === 'object') {
+        this.addFieldValidationFunctions(fields);
       }
     }
   }
 
+  private addFieldValidationFunctions(fields: { [key: string]: FieldValidationConstraint[] }) {
```

Comment:
Should this be addFieldsValidationFunctions ?

To follow SRP could it be better to implement something like (I don't like the naming I'm using):

```javascript
private addFieldsValidationFunctions(...) {
  for(let field in fields) {
       addFieldValidationFunctions(...)
  }
}

private addFieldValidationsFunction() {
   const fieldValidationConstraints = fields[field];
   if (fieldValidationConstraints instanceof Array) {
       addFieldValdationFunction(...) 
       }
  }
(...)
}
```